File: Algoritmos/Aula #03/aula3.py
def merge(lis1, tam_lista1, list2, tam_lista2):
    contador1 = 0
    contador2= 0
    lista_aux =[]
    while True:
        if contador1 == tam_lista1:
            while True:
                lista_aux.append(list2[contador2])
                contador2 += 1
                if contador2 == tam_lista2:
                    return lista_aux
                
        if contador2 == tam_lista2:
            while True:
                lista_aux.append(lis1[contador1])
                contador1 += 1
                if contador1 == tam_lista1:
                    return lista_aux
                
        if lis1[contador1] < list2[contador2]:
            lista_aux.append(lis1[contador1])
            contador1 += 1
        else:
            lista_aux.append(list2[contador2])
            contador2 += 1

            

def mergesort(lista, tam_lista):
    if tam_lista == 1:
        return lista
    if tam_lista %2 == 0:
        return merge(mergesort(lista[:(tam_lista//2)],tam_lista//2),tam_lista//2, mergesort(lista[(tam_lista//2):], (tam_lista//2)), tam_lista//2)
    else:
        return merge(mergesort(lista[:(tam_lista//2)],tam_lista//2), tam_lista//2, mergesort(lista[(tam_lista//2):], 1+(tam_lista//2)), 1+tam_lista//2)


def convert_lista(lista):
    for i in range(len(lista)):
        lista[i] = int(lista[i])
    return lista
    
    
lista = [3,45,65,9,8,114,2,1,0,3,3,99,4,3,3,45,6,47,9]
lista = mergesort(lista, len(lista))
print(*lista)

File: Algoritmos/Aula #03/aula3.c
#include <stdio.h>

int merge(int lis1[], int tam_lista1, int list2[], int tam_lista2){
    int contador1 = 0;
    int contador2= 0;
    int tam_array = tam_lista1 + tam_lista2;
    int lista_aux[tam_array];
    int cont_array = 0;
    while (1){

        if (contador1 == tam_lista1){
            while (1){
                lista_aux[cont_array] = list2[contador2];
                contador2 = contador2 + 1;
                cont_array = cont_array + 1;
                if (contador2 == tam_lista2){
                    return lista_aux;
                    }
                }
            }
                
        if (contador2 == tam_lista2){
            while (1){
                lista_aux[cont_array] = lis1[contador1];
                contador1 = contador1 + 1;
                cont_array = cont_array + 1;
                if (contador1 == tam_lista1){
                    return lista_aux;
                    }
                }
            }
                
        if (lis1[contador1] < list2[contador2]){
                lista_aux[cont_array] = lis1[contador1];
                contador1 = contador1 + 1;
                cont_array = cont_array + 1;
            }
        else{
            lista_aux[cont_array] = list2[contador2];
            contador2 = contador2 + 1;
            cont_array = cont_array + 1;
            }
    }
}







void print_array(int array[], int tam_array){
    for (int i = 0; i < tam_array; i++){
        printf("%d ", array[i]);
    }
}


int main(){
    int numero_de_casos;
    scanf("%d",&numero_de_casos);
    for (int caso=0; caso<numero_de_casos; caso++) {
        int tam_array;
        scanf("%d" ,&tam_array);
        int array[tam_array];
        for (int j=0; j<tam_array; j++){
            scanf("%d" ,&array[j]);
        }

    }
    
    return 0;
}